projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -482,6 +495,9 @@ public static int GetFieldValueByteCount(object value)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int GetUTF8ByteCount(string val) => string.IsNullOrEmpty(val) ? 0 : Encoding.UTF8.GetByteCount(val);
```

Comment:
Did you test `GetUTF8ByteCount(ROS<char> val) => val.Length == 0 ? 0 : Encoding.UTF8.GetByteCount(val)`? span's have less error checks and might be a tiny bit faster

Reply:
Added it as it was a tiny bit faster. Thanks for the tip.